BUILDER_DEV.md

```diff
@@ -1,390 +1,285 @@
-You need to make sure you have the required packages installed.
-Run the appropriate shell script under the `support/linux` folder to install the packages.
-Refer to [BUILDING.md](./BUILDING.md) doc for the detailed steps.
+There are several ways of creating a Builder dev environment - but supporting all operating systems and environments has proven to be untenable. This document includes one officially supported way of creating a Builder dev environment, and links to unsupported ways of creating the dev environment that you may use at your own risk.
 
-## Create Builder encryption key pair
-If you haven't already done this earlier, you'll need to create a key pair for builder to encrypt credentials.
-Do this via the `hab user key generate bldr` command. This will generate a key pair with the prefix `bldr`
-in your local `hab/cache/keys` directory. Copy the key pair over to some location that is accessible by the running services.
-You will need to specify that location in the config files below.
+Eventually, we will want to harness the power of the Habitat studio to automate a builder dev environment, but this serves as a stop gap for now.
 
-## Create configuration files
-Some capabilities (such as allowing builder permissions, and turning on auto-building when new packages are uploaded to the depot) require passing in a custom config to the Builder services.
+## Officially Supported Dev Environment
 
-Create the following files somewhere on your local filesystem:
+This Dev Environment involves spinning up a virtual machine (most of the core contributors use VMWare, but Virtual Box should work as well) running Ubuntu 17.10 desktop.
 
-`config_api.toml`
-```toml
-[depot]
-path = "/hab/svc/builder-api/data"
-key_dir = "/path/to/bldr-key-pair"
+### Pre-Reqs
+
+* Ubuntu 17.10 Desktop Virtual Machine
+* Access to the "core" origin in [Production Builder](https://bldr.habitat.sh). Ask a core maintainer (or in the [Habitat Slack](http://slack.habitat.sh/)) if you do not have access.
+* Access to the Habitat 1Password account
+* All of the following steps should be run within your Virtual Machine
+* A [Github Personal Access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) with all repo and all user permissions.
+* The [Habitat Builder App](https://github.com/apps/habitat-builder) installed on your Github account.
+* Open your Sudoers file at /etc/sudoers and remove the "secure_path" section (this is required for the make scripts to run correctly)
+
+### Builder Setup
+* Update your system
+```
+$ sudo apt-get update
+```
+
+* Install git
+```
+$ sudo apt-get install git
+```
+
+* Clone the [Habitat repo](https://github.com/habitat-sh/habitat)
+```
+$ git clone https://github.com/habitat-sh/habitat.git
+```
+
+* Install the [Habitat CLI](https://www.habitat.sh/docs/get-habitat/)
+* Copy habitat-builder-dev.2017-10-02.private-key.pem from 1Password into path/to/habitat/repo/.secrets/builder-github-app.pem
```

Comment:
I understand this guide is mainly for core contributors. For those who haven't access to secrets (like me ;)). How can we have our own secret, does that mean we'd need to fork the github app? (is it even possible?)

Reply:
There is some work in progress by our partners at Kinvolk on this - will definitely update the docs when it is ready to be merged https://github.com/habitat-sh/habitat/pull/4161